tendrl/commons/objects/cluster/atoms/import_cluster/__init__.py

```diff
@@ -184,6 +184,13 @@ def run(self):
                             job_id=self.parameters['job_id'],
                             flow_id=self.parameters['flow_id']
                         )
+                        for child_job_id in parent_job.children:
+                            child_job = NS.tendrl.objects.Job(
+                                job_id=child_job_id
+                            ).load()
+                            if child_job.status not in ["finished", "failed"]:
+                                child_job.status = "failed"
```

Comment:
please add comment here explaining the status check and change

Reply:
Child jobs in Import and Unmanage remain in new state if parent job gets timed out and failed. Child jobs should be marked failed explicitly and should not be waited for ttl to mark them failed. These stale jobs are picked up by node-agent in a scenario in which an import is fired but one of the nodes is down. After import failure if the node comes up, it will pick up those jobs and keep the node-agent busy and defer other tasks such as Unmanage( if fired immediately after import failure) which may fail them due to timeout.

Reply:
https://github.com/anmolsachan/commons/blob/7ff3d2ece3f92c5c963ddce0299bb41dc86c5b2f/tendrl/commons/objects/cluster/atoms/import_cluster/__init__.py#L181 move this logic inside timout block, when parent timeout then marks child job as fail